packages/datagateway-common/src/setupTests.tsx

```diff
@@ -15,6 +15,8 @@ import thunk from 'redux-thunk';
 import configureStore from 'redux-mock-store';
 import { createMemoryHistory, History } from 'history';
 
+jest.setTimeout(15000);
```

Comment:
What does this `setTimeout` here do? Does it set the timeout of every test in `datagateway-common` to be 15 seconds? If you could add a comment explaining why it is needed here that would be very helpful.

Reply:
I figured we occasionally have tests failing due to timeout issues, and then people add longer timeouts to those specific tests. I just kinda figured increasing the global timeout is fine - as in if a test fails it'll just take longer to fail (if it's an async) test, and it sorts out any unstable tests that are hard to figure out because the reason they're failing is due to timeouts which depend on the system at the time of running.

I could theoretically flag to only increase the timeouts in CI, but I figured it's not bad to just increase them generally for all packages.

Reply:
I see. I do wonder why tests take longer than 5 seconds sometimes. I know JSDOM isn't exactly the fastest, but it can't be that slow to the point where some tests can take more than 5 seconds to run. It just seems excessive to me.

Reply:
I think the download tests specifically which were the ones that were failing often are as I detailed in the commit, failing because the tests are doing a lot in one test. e.g. in the tests they're typing in filters 2/3/4 times, and `user.type` simulates every keystroke, so potentially if stuff rerenders when the filters change that's a lot of stuff to do in 5 seconds

As I said, the alternative is to refactor those tests potentially, but there will likely always be tests that can occasionally breach 5 secs, and do we want that to then fail our CI?

Reply:
Also, as a side effect this will help those of us who have terrible laptops (🙋‍♀️) - sometimes when running too many tests at once tests that don't normally time out do - so the global increase timeout helps as well 😅

Reply:
Good point! I think the code is pretty self-explanatory, so I don't think adding comment is necessary.